contrib/pkg/adm/managedns/enable.go

```diff
@@ -0,0 +1,191 @@
+)
+
+var (
+	validClouds = map[string]bool{
+		cloudAWS: true,
+		// TODO:
+		//cloudGCP:   true,
+	}
+)
+
+// Options is the set of options to generate and apply a new cluster deployment
+type Options struct {
+	Cloud     string
+	CredsFile string
+	homeDir   string
+}
+
+// NewEnableManageDNSCommand creates a command that generates and applies artifacts to enable managed
+// DNS globally for the Hive cluster.
+func NewEnableManageDNSCommand() *cobra.Command {
+	opt := &Options{}
+
+	cmd := &cobra.Command{
+		Use:   `enable domain1.example.com domain2.example.com ...`,
+		Short: "Enable managed DNS globally for the Hive cluster.",
+		Long:  longDesc,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			if err := opt.Complete(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Validate(cmd); err != nil {
+				return
+			}
+			dynClient, err := contributils.GetClient()
+			if err != nil {
+				log.WithError(err).Fatal("error creating kube clients")
+			}
+
+			err = opt.Run(dynClient, args)
+			if err != nil {
+				log.WithError(err).Error("Error")
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opt.Cloud, "cloud", cloudAWS, "Cloud provider: aws(default)|gcp)")
+	flags.StringVar(&opt.CredsFile, "creds-file", "", "Cloud credentials file (defaults vary depending on cloud)")
+	return cmd
+}
+
+// Complete finishes parsing arguments for the command
+func (o *Options) Complete(cmd *cobra.Command, args []string) error {
+	o.homeDir = "."
+
+	if u, err := user.Current(); err == nil {
+		o.homeDir = u.HomeDir
+	}
+	return nil
+}
+
+// Validate ensures that option values make sense
+func (o *Options) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run executes the command
+func (o *Options) Run(dynClient client.Client, args []string) error {
+	if err := apis.AddToScheme(scheme.Scheme); err != nil {
+		return err
+	}
+	rh, err := o.getResourceHelper()
+	if err != nil {
+		return err
+	}
+
+	// Update the current HiveConfig, which should always exist as the operator will
+	// create a default one once run.
+	hc := &hivev1.HiveConfig{}
+	err = dynClient.Get(context.Background(), types.NamespacedName{Name: "hive"}, hc)
+	if err != nil {
+		log.WithError(err).Fatal("error looking up HiveConfig 'hive'")
+	}
+
+	hc.Spec.ManagedDomains = args
+
+	var credsSecret *corev1.Secret
+
+	if o.Cloud == cloudAWS {
+		// Apply a secret for credentials to manage the root domain:
+		credsSecret, err = o.generateAWSCredentialsSecret()
+		if err != nil {
+			log.WithError(err).Fatal("error generating manageDNS credentials secret")
+		}
+		accessor, err := meta.Accessor(credsSecret)
+		if err != nil {
+			log.WithError(err).Errorf("Cannot create accessor for object of type %T", credsSecret)
+			return err
+		}
+		accessor.SetNamespace(hiveNamespace)
+		rh.ApplyRuntimeObject(credsSecret, scheme.Scheme)
+	}
+
+	hc.Spec.ExternalDNS = &hivev1.ExternalDNSConfig{
```

Comment:
This should be inside the `if o.Cloud == cloudAWS` statement. It would then be nice to use `credsSecret.Name` instead of `manageDNSCredentialsSecret`.